### PR TITLE
Complete wsgiref

### PR DESCRIPTION
**Pull request by srittau**
* Add `wsgiref.handlers` and `wsgiref.simple_server`
* Add `StartResponse` to `wsgiref.types`

### CODE REVIEW

File: stdlib/2and3/wsgiref/simple_server.pyi [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+sys_version: str  # undocumented
+software_version: str  # undocumented
+
+class ServerHandler(SimpleHandler):  # undocumented
+    server_software: str
+    def close(self) -> None: ...
+
+class WSGIServer(HTTPServer):
+    application: Optional[WSGIApplication]
+    base_environ: Dict[str, str]  # only available after call to setup_environ()
```
Comment (JelleZijlstra) [inline stdlib/2and3/wsgiref/simple_server.pyi]: Shouldn't this be WSGIEnvironment?

File: stdlib/2and3/wsgiref/handlers.pyi [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    def close(self) -> None: ...
+    def send_headers(self) -> None: ...
+    def result_is_file(self) -> bool: ...
+    def client_is_modern(self) -> bool: ...
+    def log_exception(self, exc_info: _exc_info) -> None: ...
+    def handle_error(self) -> None: ...
+    def error_output(self, environ: WSGIEnvironment, start_response: StartResponse) -> List[bytes]: ...
+
+    def _write(self, data: bytes) -> None:
+        raise NotImplementedError()
```
Comment (JelleZijlstra) [inline stdlib/2and3/wsgiref/handlers.pyi]: I don't think this does anything in stubs. Maybe we should decorate these with `@abc.abstractmethod` instead.

File: stdlib/2and3/wsgiref/handlers.pyi [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    def get_stdin(self) -> InputStream: ...
+    def get_stderr(self) -> ErrorStream: ...
+    def add_cgi_vars(self) -> None: ...
+    def _write(self, data: bytes) -> None: ...
+    def _flush(self) -> None: ...
+
+class BaseCGIHandler(SimpleHandler): ...
+
+class CGIHandler(BaseCGIHandler):
+    os_environ: MutableMapping[str, str]
```
Comment (JelleZijlstra) [inline stdlib/2and3/wsgiref/handlers.pyi]: This is redundant since it's the same on the base class.

**Comment by srittau** [comment]
I need guidance with the last declaration in `wsgiref.simple_server`. Currently it reads:

```python
_S = TypeVar("_S", bound=WSGIServer)

def make_server(host: str, port: int, app: WSGIApplication, server_class: Type[_S] = ..., handler_class: Type[WSGIRequestHandler] = ...) -> _S: ...
```

Of course that means that calling `make_server("localhost", 80, app)` type checkers will not be able to infer the return type. So I tried using this:

```python
_S = TypeVar("_S", bound=WSGIServer)

@overload
def make_server(host: str, port: int, app: WSGIApplication, *, handler_class: Type[WSGIRequestHandler] = ...) -> WSGIServer: ...
@overload
def make_server(host: str, port: int, app: WSGIApplication, server_class: Type[_S] = ..., handler_class: Type[WSGIRequestHandler] = ...) -> _S: ...
```

Unfortunately mypy then complains:

```
stdlib/2and3/wsgiref/simple_server.pyi:38: error: Overloaded function signatures 1 and 2 overlap with incompatible return types
```

Is there any solution?

**Pull request merged**